compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/model/types/BUnionType.java

```
@@ -84,4 +85,39 @@ public String getDesc() {
     public void setNullable(boolean nullable) {
         this.nullable = nullable;
     }
+
+    @Override
+    public boolean hasImplicitInitialValue() {
+        // NIL is a member.
+        if (memberTypes.stream().anyMatch(t -> t.tag == TypeTags.NIL)) {
+            return true;
+        }
+
+        // All members are of same type and has the implicit initial value as a member.
+        if (!memberTypes.isEmpty()) {
```

Shall we invert the if condition and fail fast?
```java
 if (memberTypes.isEmpty()) { 
    return false;
 }
```